serde/src/java/org/apache/hadoop/hive/serde2/objectinspector/StandardConstantMapObjectInspector.java

```diff
@@ -40,8 +41,10 @@ protected StandardConstantMapObjectInspector() {
    */
    protected StandardConstantMapObjectInspector(ObjectInspector mapKeyObjectInspector,
        ObjectInspector mapValueObjectInspector, Map<?, ?> value) {
-    super(mapKeyObjectInspector, mapValueObjectInspector);
-    this.value = value;
+     super(mapKeyObjectInspector, mapValueObjectInspector);
+     if (null != value) {
```

Comment:
please remove this and similar conditionals.... a map which is `NULL` is not `EMPTY`!
this should be fixed in `StatsUtils`